src/regdesk/states/reg_machine_args.js

```diff
@@ -0,0 +1,49 @@
+import { WorkflowButton } from '../workflow_button.js';
+
+/**
+ * Record for the common state machine args for states.
+ */
+export class RegMachineArgs {
+  /**
+   * Initializes a new instance of the RegMachineArgs class.
+   * @param {WorkflowButton} leftButton - The left side gutter button.
+   * @param {WorkflowButton} rightButton - The right side gutter button.
+   * @param {Element} centerField - The center container for page content.
+   * @param {PrinterManager} printerManager - The printer manager that manages printers.
+   * @param {CommunicationManager} commManager - The communication manager for making API calls.
+   */
+  constructor(
+    leftButton,
+    rightButton,
+    centerField,
+    printerManager,
+    commManager,
+  ) {
+    this.leftButton = leftButton;
+    this.rightButton = rightButton;
+    this.centerField = centerField;
+    this.printerManager = printerManager;
+    this.commManager = commManager;
+
+    if (this.constructor === RegMachineArgs.constructor) {
+      Object.freeze(this);
```

Comment:
:D! Nice! I love the immutability!

Reply:
I stumbled across it when looking at some library that had a whole bunch of different ways of creating / generating various types of immutable things. The constructor guard makes sure it's _this_ constructor we called, as opposed to an extending class which should be doing its own freezing in its own constructor.